src/git-sync

```diff
@@ -26,7 +29,11 @@ function preconditions {
   if [ "$1" = "--all" ]; then
     sync_all=true
   elif [ "$(is_feature_branch "$INITIAL_BRANCH_NAME")" = false ]; then
+    # syncing a non-feature branch
```

Comment:
I am going with a single variable `sync_target` now, which is set to either "all branches", "feature branch", or "non-feature branch". I think that makes the code self-describing enough to remove the comments, and also nicely dries up the checks, thereby making this code faster. Great idea!